python/pyarrow/ipc.pxi

```diff
@@ -437,7 +437,11 @@ cdef class MessageReader(_Weakrefable):
 
     def __iter__(self):
         while True:
-            yield self.read_next_message()
+            try:
+                yield self.read_next_message()
+            except StopIteration:
+                # For Cython >= 3.0.0
+                return
```

Comment:
I don't know if it is this code that gives problems to be written in a way that works for both cython 0.29 and 3, but a potential different way of writing it:

```
def __iter__(self):
    return self

def __next__(self):
    return self.read_next_message()
```

Reply:
Thanks! I'll try it.